scripts/src/workflowtesting/checkprforci.py

```diff
@@ -0,0 +1,84 @@
+import re
+import argparse
+import os
+import requests
+import json
+import yaml
+try:
+    from yaml import CLoader as Loader, CDumper as Dumper
+except ImportError:
+    from yaml import Loader, Dumper
+
+def check_if_ci_only_is_modified(api_url):
+    # api_url https://api.github.com/repos/<organization-name>/<repository-name>/pulls/1
+
+    files_api_url = f'{api_url}/files'
+    headers = {'Accept': 'application/vnd.github.v3+json'}
+    pattern_workflow = re.compile(r".github/workflows/.*")
+    pattern_script = re.compile(r"scripts/.*")
+    pattern_test = re.compile(r"tests/.*")
+    page_number = 1
+    max_page_size,page_size = 100,100
+
+    workflow_found = False
+    other_found = False
+    test_config_path = ""
+
+    while (page_size == max_page_size) & (not other_found):
+
+        files_api_query = f'{files_api_url}?per_page={page_size}&page={page_number}'
+        r = requests.get(files_api_query,headers=headers)
+        files = r.json()
+        page_size = len(files)
+        page_number += 1
+
+
+        for f in files:
+            filename = f["filename"]
+            if pattern_workflow.match(filename):
+                workflow_found = True
+            elif pattern_script.match(filename):
+                workflow_found = True
+            elif pattern_test.match(filename):
+                workflow_found = True
+            else:
+                other_found = True
+                break
```

Comment:
Instead of break, we can early return here..

Reply:
changed to early return - so second condition removed from while condition above.